### PR TITLE
Merge sample types and object types

### PR DESCRIPTION
**Pull request by marikoja**

### CODE REVIEW

File: frontend/src/components/users/ShowUsers.jsx
Comment (bjkeller) [inline frontend/src/components/users/ShowUsers.jsx:127]: Avoid use of permission values in constants  since they are configurable in the backend

**Review by bjkeller — CHANGES_REQUESTED** [review]
Please remove the explicit use of the permission constant `.6`

File: frontend/src/components/users/ShowUsers.jsx
Comment (gnomicosuw) [reply frontend/src/components/users/ShowUsers.jsx]: This is done and pushed. However, something happened in your commit

014eba0d86da2c8fb6477972edc5195916262bcc "Merge branch 'v3-refactor' into merge_samples_objects"

that blew up eslint.  The code now does not compile in the sample_types area. I will leave it to you and Mariko to clean up since I don't know what else may or may not have changed...(If I just fix the eslint stuff I may be missing other logic that was in the branch before the merge)

Here is the full list:

src/components/sampleTypes/fields/SampleTypeFieldForm.spec.js
  Line 3:1:     Expected indentation of 2 spaces but found 1  indent
  Line 3:10:    'fireEvent' is defined but never used         no-unused-vars
  Line 3:38:    'wait' is defined but never used              no-unused-vars
  Line 3:51:    Trailing spaces not allowed                   no-trailing-spaces
  Line 18:24:   A space is required after '{'                 object-curly-spacing
  Line 18:41:   A space is required before '}'                object-curly-spacing
  Line 18:44:   A space is required after '{'                 object-curly-spacing
  Line 18:61:   A space is required before '}'                object-curly-spacing
  Line 35:7:    Missing semicolon                             semi
  Line 42:42:   A space is required after '{'                 object-curly-spacing
  Line 42:55:   A space is required before '}'                object-curly-spacing
  Line 105:1:   Expected indentation of 4 spaces but found 3  indent
  Line 106:1:   Expected indentation of 6 spaces but found 5  indent
  Line 108:1:   Expected indentation of 6 spaces but found 5  indent
  Line 109:1:   Expected indentation of 4 spaces but found 3  indent
  Line 111:1:   Expected indentation of 4 spaces but found 3  indent
  Line 112:1:   Expected indentation of 6 spaces but found 5  indent
  Line 113:1:   Expected indentation of 6 spaces but found 5  indent
  Line 115:1:   Expected indentation of 6 spaces but found 5  indent
  Line 117:1:   Expected indentation of 6 spaces but found 5  indent
  Line 118:1:   Expected indentation of 4 spaces but found 3  indent
  Line 120:1:   Expected indentation of 4 spaces but found 3  indent
  Line 121:1:   Expected indentation of 6 spaces but found 5  indent
  Line 122:1:   Expected indentation of 6 spaces but found 5  indent
  Line 124:1:   Expected indentation of 6 spaces but found 5  indent
  Line 126:1:   Expected indentation of 6 spaces but found 5  indent
  Line 127:1:   Expected indentation of 4 spaces but found 3  indent
  Line 160:10:  Block must not be padded by blank lines       padded-blocks
  Line 162:1:   More than 1 blank line not allowed            no-multiple-empty-lines
  Line 290:8:   Missing semicolon                             semi
  Line 302:42:  A space is required after '{'                 object-curly-spacing
  Line 302:58:  A space is required before '}'                object-curly-spacing
  Line 351:41:  A space is required after '{'                 object-curly-spacing
  Line 351:62:  A space is required before '}'                object-curly-spacing
  Line 368:9:   Missing semicolon                             semi
  Line 370:5:   Missing semicolon                             semi

src/components/shared/Buttons.spec.js
  Line 2:8:    A space is required after '{'            object-curly-spacing
  Line 2:34:   A space is required before '}'           object-curly-spacing
  Line 2:65:   Missing semicolon                        semi
  Line 5:8:    A space is required after '{'            object-curly-spacing
  Line 5:35:   A space is required before '}'           object-curly-spacing
  Line 19:48:  A space is required after '{'            object-curly-spacing
  Line 19:66:  A space is required before '}'           object-curly-spacing
  Line 39:30:  A space is required after ','            comma-spacing
  Line 39:40:  Block must not be padded by blank lines  padded-blocks

src/components/shared/DropdownMenu.spec.js
  Line 1:8:   'React' is defined but never used         no-unused-vars
  Line 2:10:  'render' is defined but never used        no-unused-vars
  Line 2:18:  'screen' is defined but never used        no-unused-vars
  Line 3:8:   'DropdownMenu' is defined but never used  no-unused-vars

src/components/shared/LoadingSpinner.spec.js
  Line 11:1:  Expected indentation of 4 spaces but found 6  indent
  Line 12:1:  Expected indentation of 4 spaces but found 6  indent
  Line 17:1:  Expected indentation of 4 spaces but found 6  indent
  Line 19:1:  Expected indentation of 4 spaces but found 6  indent
  Line 20:1:  Expected indentation of 4 spaces but found 6  indent
  Line 21:1:  Expected indentation of 4 spaces but found 6  indent

src/components/shared/Modal.spec.js
  Line 3:1:   Expected indentation of 2 spaces but found 1  indent
  Line 3:37:  Trailing spaces not allowed                   no-trailing-spaces
  Line 15:1:  More than 1 blank line not allowed            no-multiple-empty-lines
  Line 28:1:  Expected indentation of 4 spaces but found 2  indent
  Line 38:8:  Block must not be padded by blank lines       padded-blocks

src/components/shared/Select.spec.js
  Line 18:4:   Missing semicolon               semi
  Line 35:11:  A space is required after '{'   object-curly-spacing
  Line 35:21:  A space is required before '}'  object-curly-spacing
  Line 54:3:   Missing semicolon               semi

src/components/shared/TextInput.spec.js
  Line 7:37:  Missing semicolon             semi
  Line 18:6:  Strings must use singlequote  quotes

**Review by bjkeller — APPROVED** [review]

**Pull request merged**
